packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/PersistedNameTests.kt

```diff
@@ -409,3 +449,20 @@ class PersistedNameChildSample : RealmObject {
     @PersistedName("persistedNameParents")
     val publicNameParents by backlinks(PersistedNameParentSample::publicNameChildField)
 }
+
+@PersistedName("PersistedParent")
+class RealmParent(var id: Int) : RealmObject {
+    constructor() : this(0)
+    var child: RealmChild? = null
+}
+
+// Should conflict with RealmParent if included in the same Realm.
+class PersistedParent(var id: Int) : RealmObject {
```

Comment:
Nitpick ... but I guess we should also have a test of clashes between two classes with `@PersistedName` with the same name.